File: BridgewellDemoObjectiveC/WebViewController.m
//
//  WebViewController.m
//  BridgewellDemoObjectiveC
//
//

#import "WebViewController.h"
#import "WebKit/WebKit.h"
#import "BridgewellSDK/BridgewellSDK.h"

@interface WebViewController ()
@property (nonatomic) WKWebView *webview;
@property (nonatomic) WKUserContentController *contentController;
@end

@implementation WebViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Let HTML videos with a "playsinline" attribute play inline.
    self.contentController = [[WKUserContentController alloc] init];
    
    WKWebViewConfiguration *webViewConfiguration = [[WKWebViewConfiguration alloc] init];
    // Let HTML videos with a "playsinline" attribute play inline.
    webViewConfiguration.allowsInlineMediaPlayback = true;
    // Let HTML videos with an "autoplay" attribute play automatically.
//    webViewConfiguration.mediaTypesRequiringUserActionForPlayback = [];
    
    self.webview = [[WKWebView alloc] initWithFrame:self.view.frame configuration:webViewConfiguration];
    [self.view addSubview:self.webview];
    
    [[Bridgewell shared] registerContentWebViewWithAdInfo:self.webview completion:^{
        [self handleLoadWebview];
    }];
}

-(void)handleLoadWebview {
    NSURL *url = [[NSURL alloc] initWithString:@"https://img.scupio.com/html/sdk-webview-test.html"];
    NSURLRequest *request = [[NSURLRequest alloc] initWithURL:url];
    [self.webview loadRequest:request];
}

@end

File: BridgewellDemo/WebViewController.swift
//
//  WebViewController.swift
//  BridgewellDemo
//
//

import UIKit
import WebKit
import BridgewellSDK

class WebViewController: UIViewController {
    var webView: WKWebView!
    
    override func viewDidLoad() {
        super.viewDidLoad()

        // Initialize a WKWebViewConfiguration object.
        let webViewConfiguration = WKWebViewConfiguration()
        // Let HTML videos with a "playsinline" attribute play inline.
        webViewConfiguration.allowsInlineMediaPlayback = true
        // Let HTML videos with an "autoplay" attribute play automatically.
        webViewConfiguration.mediaTypesRequiringUserActionForPlayback = []
        // Initialize the WKWebView with your WKWebViewConfiguration object.
        #if DEBUG
        webViewConfiguration.preferences.setValue(true, forKey: "developerExtrasEnabled")
        #endif
        webView = WKWebView(frame: view.frame, configuration: webViewConfiguration)
        webView.translatesAutoresizingMaskIntoConstraints = false
        view.addSubview(webView)
        NSLayoutConstraint.activate([
            webView.topAnchor.constraint(equalTo: view.safeAreaLayoutGuide.topAnchor),
            webView.leftAnchor.constraint(equalTo: view.safeAreaLayoutGuide.leftAnchor),
            webView.rightAnchor.constraint(equalTo: view.safeAreaLayoutGuide.rightAnchor),
            webView.bottomAnchor.constraint(equalTo: view.safeAreaLayoutGuide.bottomAnchor)
        ])
        
        // Register webview to inject extra properties to html global variable
        // All of them are optional, and retrieved from device
        Bridgewell.shared.registerContentWebViewWithAdInfo(webView) {[weak self] in
            // Callback here
            // Handle after success register some properties to webview
            guard let `self` = self else { return }
            self.handleLoadWebview()
        }
        
        setNavbarLogo()
        setBackBarButton()
    }
    
    private func handleLoadWebview() {
        webView.load(URLRequest(url: URL(string: "https://img.scupio.com/html/sdk-webview-test.html")!))
    }

}
